tests/utils/palette_review.py

```diff
@@ -0,0 +1,96 @@
+import cssutils
+import logging
+from matplotlib import colors
+import os
+from pathlib import Path
+from requests import get, exceptions
+
+cssutils.log.setLevel(logging.ERROR)
+palette_dir = Path(os.getcwd()) / "terminal/css/palettes"
+palettes = [p for p in palette_dir.glob(pattern="*.css") if "default" not in p.name]
+color_labels = colors.cnames.keys()
+bg_labels = ["background", "bg"]
+fg_labels = ["foreground", "fg"]
+drop_labels = ["var(", "invert"]
+palettes.sort()
+
+intro = ('## Steps\n'
+         '1. Identify the background and foreground colors '
+         'based on simple keywords (ex: property name contains "background")\n'
+         '2. Check each background-foreground combo for WCAG alignment\n'
+         '3. Report results in an html table\n'
+         '## Notes\n'
+         '- ___keywords___: bg, background, fg, foreground, color, matplotlib '
+         'color names\n'
+         '- Currently assumes all non-background colors are foreground colors for the test)\n')
+legend = ("## Legend\n"
+          "- ✅: all WCAG tests passed\n"
+          "- ☑️: some WCAG tests passed\n"
+          "- ❌: no WCAG tests passed\n"
+          "- Symbols link to WCAG api response with tests")
+
+with open(Path(os.getcwd()) / "tests" / "palette_report.md", "w") as f:
+    f.write(intro)
+    f.write(legend)
+for p in palettes:
+    backgrounds = {}
+    foregrounds = {}
+    sheet = cssutils.parseFile(p)
+    for rule in sheet:
+        if isinstance(rule, cssutils.css.csscomment.CSSComment):
+            continue
+        style_text = rule.style.cssText.replace("\n", "")
+        for i in style_text.split(";"):
+            pair = i.split(": ")
+            label = pair[0].replace("--", "")
+            value = pair[1].replace("#", "")
+            if "invert" in label or "var(" in value:
+                pass
+            else:
+                if value == "fff":
+                    value = "ffffff"
+                if any(x in label for x in bg_labels):
+                    backgrounds[label] = value
+                else:
```

Comment:
can you explain more what is going on this else block (54-60)?  i am a bit lost

how do we end up making use of the `color_labels` list from matplotlib? 
i tried out the Web AIM API with a hex value and one of the matplotlib color names and the result doesn't make sense to me

for example when i compare beige to [burlywood](https://www.colorhexa.com/deb887)

```https://webaim.org/resources/contrastchecker/?fcolor=faf0e6&bcolor=burlywood&api```

where beige hex is `#faf0e6` i get all passes even though by my eye they seem very similar:

```json
{"ratio":"18.4","AA":"pass","AALarge":"pass","AAA":"pass","AAALarge":"pass"}

```

Reply:
comparing beige to the burlywood hexcode `#deb887` fails:

```https://webaim.org/resources/contrastchecker/?fcolor=faf0e6&bcolor=deb887&api```:

```json
{"ratio":"1.65","AA":"fail","AALarge":"fail","AAA":"fail","AAALarge":"fail"}
```

so if we are using the matplotlib color label i think we need to look up the hex code first 
(again, wasn't entirely sure just from reading)

Reply:
The else block on lines 54-60 is just a basic way to filter for foreground labels or color labels (treated as foreground) within the palettes. This is the only place where the matplotlib color labels are used currently. I just added that as a hacky way to capture additional color labels from palettes, assuming that new palettes could be imported any time. 

So you are correct that the review should only be using hex codes and I think that it will only do so as long as the palette css files follow the existing structure (_color-name: #HEXVALUE_).

Reply:
Also we can definitely remove the two elif statements if it is decided that all palette css files will follow the same naming convention, using "fg"/"foreground" and "bg"/"background" in the color labels. Other color types (ex: accents) could also be defined as needed. 

Right now some palettes may include words like "primary", "secondary", tertiary", "color", or an actual color name (ex: "red") in the label instead of the foreground/background strings, so that is why the elif statements on lines 57-60 are currently used.